extensions/ql-vscode/src/stories/variant-analysis/VariantAnalysisAnalyzedRepos.stories.tsx

```diff
@@ -125,24 +125,22 @@ Example.args = {
 };
 
 faker.seed(42);
-const uniqueStore = {};
+const uniqueNumbers = [...Array(1000).keys()].sort(() => Math.random() - 0.5);
 
 const manyScannedRepos = Array.from({ length: 1000 }, (_, i) => {
   const mockedScannedRepo = createMockScannedRepo();
+  const randomInt = uniqueNumbers.pop();
+  const uniqueId =
+    randomInt === undefined ? Math.random() * 8000 + 1001 : randomInt;
```

Comment:
Since array.pop() returns a number OR undefined I needed to add an alternative for the undefined case. This is a best effort for a random number bigger than 1000 without using a library to create it. This case should never happen since the uniqueNumbers array is never undefined.

Reply:
I'm not a huge fan of having code that's not intended to execute, and that branch isn't just throwing an error. I have a few potential solutions to make things a little cleaner:

### Throw an error instead

I'd be happy if we threw an error if `pop` ever returns `undefined`. As you say it shouldn't ever happen, so if it does then the code needs to be updated rather than using a fallback strategy.

### Iterate over `uniqueNumbers`

Instead of doing `Array.from({ length: 1000 }, (_, i) =>` to initiate the loop, do `uniqueNumbers.map(repoId =>` so that we remove the need to call `pop`.

### Just use `Math.random()` and don't precompute a random list

The repo ids don't have to go from 0-999. They just need to be unique, and even then it's only the storybook so not too critical. If we use big enough numbers, e.g. `Math.floor(Math.random() * 1000000000000)` (AFAIK the max safe int in javascript is `9007199254740991`), then the chance of getting duplicates in our 1000 repo ids becomes very low.

Reply:
I think I prefer the error, as you say it should never happen and if it does we should look at it!